controllers/componentdetectionquery_controller.go

```diff
@@ -47,9 +56,101 @@ type ComponentDetectionQueryReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := r.Log.WithValues("ComponentDetectionQuery", req.NamespacedName)
+	log.Info(fmt.Sprintf("Starting reconcile loop for %v", req.NamespacedName))
+
+	// Fetch the ComponentDetectionQuery instance
+	var componentDetectionQuery appstudiov1alpha1.ComponentDetectionQuery
+	err := r.Get(ctx, req.NamespacedName, &componentDetectionQuery)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// If there is no component list in the map, the CR was just created
+	if len(componentDetectionQuery.Status.ComponentDetected) == 0 {
```

Comment:
Are there any potential scenarios where len(ComponentDetected) is 0, but we wouldn't want to go through this loop again? 

Like, say we exit out to do to some error.

Reply:
hmm interesting, yeah good point

wouldnt that fall under update in that case if we want to resubmit CR? I Wonder if its even possible. Consider this scenario:

- Submit a query CR
- No devfile, CR errors out
- user updates repo outside in git
- CR to recheck but no data is going to change, so will the server even take the request?

Reply:
Hmm, good point actually.